Remove C code from the dft_tools project

harrisonlabollita opened the pull request:
## Summary

This PR ports the C++ code in dft_tools to pure Python. Specifically, it replaces the linear tetrahedron integration routine from the VASP converter—used to calculate the density of states—with a Python implementation. Since this routine is not performance-critical, a Python version improves maintainability without impacting efficiency.

As a result, the dft_tools repository is now a Python + Fortran (via f2py) project.

## Key Changes
- Removed legacy C/C++ code from ``c++/triqs_dft_tools/converters/vasp/``.
- Reimplemented the linear tetrahedron integration routines in Python at ``python/triqs_dft_tools/converters/vasp/plovasp/lintetra.py``
- Updated the corresponding linear tetrahedron tests.

Wentzell committed:
Adjust app4triqs skeleton for python-only projects

harrisonlabollita committed:
replace linear tetrahedron c code with python impl

the-hampel committed:
[build] remove export cmake command since no c++ code is wrapped anymore

the-hampel committed:
[bug] fix used module for dos_tetra_weights_3d

the-hampel committed:
[feat] optimize tetrahedron integration for numpy

the-hampel commented on python/triqs_dft_tools/CMakeLists.txt:
```diff
@@ -3,7 +3,7 @@ configure_file(version.py.in version.py)
 # All Python files. Copy them in the build dir to have a complete package for the tests.
 file(GLOB_RECURSE python_sources RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *.py)
-file(GLOB_RECURSE wrap_generators RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *_desc.py)
+#file(GLOB_RECURSE wrap_generators RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} *_desc.py)
```
to be cleaned up

the-hampel commented on test/CMakeLists.txt:
```diff
@@ -1,4 +1,4 @@
-add_subdirectory(c++)
+#add_subdirectory(c++)
```
to be removed

the-hampel reviewed (approved):
Hi, this looks good to me. I added a numpy vectorized form of the tetrahedron. In my tests this is only a little slower than the C implementation. For 9 bands (5d orbitals NiO) and a 11x11x11 mesh of kpoints (1331) on 3001 grid points I get on my m3 macbook:  
```
33x/log: Elapsed time: 15.8411 seconds       
rewrite_py/log: Elapsed time: 24.3028 seconds
```
for the integration call. Given that this is only used to verify a setup, and should also greatly improve when symmetries are used I think we can safely replace the C code to make dft_tools python only. We can also leverage trivially mpi parallelism if we wish in the calling function in the future.

@opeil any objections? 

We should maybe enforce Python support then in this build @Wentzell does this make sense? Currently there is still some code like this: 
```
if(PythonSupport)
 add_subdirectory(python)
 add_subdirectory(python/plovasp)
 add_subdirectory(python/elk)
endif()
```
in the cmake files.

Best,
Alex

the-hampel committed:
[build] remove commented c++ cmakelists commands

the-hampel committed:
Remove C++ support and cpp2py dependency

C++ code has been completely removed from the codebase. This commit
removes all related CMake configuration:

- Remove PythonSupport option (Python is now always built)
- Remove cpp2py dependency from deps/CMakeLists.txt
- Remove cpp2py module building logic from python/triqs_dft_tools/
- Remove wrap_generators handling (no *_desc.py files exist)
- Update project description to remove cpp2py reference

The project now builds Python-only components without conditional
C++ support checks.

the-hampel commented:
I removed now all C++ dependencies in cmake and also rebased this branch on top of dft_tools/unstable (was previously 3.3.x) . Can you have a look @Wentzell if I removed the right things (I removed also dependency on cpp2py). Thanks!

Wentzell reviewed (approved):
I pushed some additional cleanups for the project structure / CMake. This looks good to be merged.
